### PR TITLE
DTSPO-12193 - Remove aksClient

### PR DESCRIPTION
**Pull request by endakelly**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-12193

### Change description ###
Remove aksClient
Jenkins agent now has tools preinstalled

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by timja — CHANGES_REQUESTED** [review]
the whole point of this is to get rid of the withDocker call here:
https://github.com/hmcts/cnp-jenkins-library/blob/d841760cfc3a8a0def31ed7405a663ef011b60ff/vars/withAcrClient.groovy#L6

you may want to keep the withAksClient so that the cluster variables get set properly.

basically the docker image is no longer needed because we install all tools on the packer agent now

File: vars/withAcrClient.groovy [hunk 1]
```diff
@@ -3,9 +3,7 @@ def call(String subscription, Closure block) {
   if (env.IS_DOCKER_BUILD_AGENT && env.IS_DOCKER_BUILD_AGENT.toBoolean()) {
     doAcrOrKubectlTask(subscription, false, block)
   } else {
-    withDocker('hmcts/cnp-aks-client:az-2.44.1-kubectl-1.26.0-helm-3.10.3', null) {
       doAcrOrKubectlTask(subscription, true, block)
```
Comment (timja) [inline vars/withAcrClient.groovy:6]: also no need to set this to true anymore as it's no longer in a container.

you should be able to remove `doAcrOrKubectlTask` as it's no longer needed, and might be able to remove the `alwaysLogin` functionality if it's no longer needed

**Review by timja — CHANGES_REQUESTED** [review]
close

File: vars/withAcrClient.groovy (hunk 1 above)
Comment (endakelly) [reply vars/withAcrClient.groovy]: Something like this? https://github.com/hmcts/cnp-jenkins-library/pull/1108/files#diff-5348ea2ed32827d22070457c3804a837f4cbbdd885a6e2d46a2939cb135294dcR2-R5

Comment (timja) [reply vars/withAcrClient.groovy]: almost, https://github.com/hmcts/cnp-jenkins-library/blob/d841760cfc3a8a0def31ed7405a663ef011b60ff/vars/withSubscriptionLogin.groovy#L8-L15 is no longer used so should be removed

File: vars/sectionBuildAndTest.groovy [hunk 2]
```diff
@@ -98,6 +98,7 @@ def call(params) {
                         # in case anyone doesn't have a trailing new line in their file
                         echo -e '\n' >> .dockerignore
                         cat .dockerignore_build >> .dockerignore
+                        sed -i 's/-e//g' .dockerignore
```
Comment (timja) [inline vars/sectionBuildAndTest.groovy:101]: what's this for?

Comment (endakelly) [reply vars/sectionBuildAndTest.groovy:101]: For some reason the docker ignore is appending `-e` to the `.jar` file. I will change it to replace `.jar-e` with `.jar` so it doesn't muck up images with `-e` in their names. I think it's something to do with the echo

Comment (timja) [reply vars/sectionBuildAndTest.groovy:101]: probably needs a bit more debugging, seems quite odd =/

**Review by timja — CHANGES_REQUESTED** [review]

File: vars/withSubscriptionLogin.groovy [hunk 3]
```diff
@@ -5,28 +5,19 @@
  * @param body the body to execute after logged in
  */
 def call(String subscription, boolean alwaysExecute, Closure body) {
```
[the remark this answers is not in the document]
Comment (timja) [reply vars/withSubscriptionLogin.groovy:7]: still present

**Review by timja — CHANGES_REQUESTED** [review]
https://github.com/hmcts/cnp-jenkins-library/pull/1108/files/d7ccceb12f1585b25738ffecfcd7fd4e44552da0#diff-1d58eed88e02bb1f6c54e9e4799c78f65c2d05a79afa269c785a1a0e135d5e94

**Review by timja — APPROVED** [review]

File: vars/withSubscriptionLogin.groovy [hunk 4]
```diff
@@ -32,4 +32,4 @@ def call(String subscription, boolean alwaysExecute, Closure body) {
 
 def call(String subscription, Closure body) {
```
Comment (endakelly) [inline vars/withSubscriptionLogin.groovy:33]: @timja the pipeline is complaining about this being a duplicate method of the same signature. Can this block be removed or should it be renamed?

Comment (timja) [reply vars/withSubscriptionLogin.groovy:33]: remove it as it's no longer needed

**Review by timja — APPROVED** [review]

**Pull request merged**